backend/app/core/background_runner.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import json
+
+from app.core.bot import BaseBotClient
+from app.core.config import settings
+
+
+class BackgroundRunner:
+    def __init__(self):
+        self.client = BaseBotClient.get_bot_client()
+        self.background_task = None
+
+    async def start_bot(self):
+        try:
+            with open(settings.matrix_bot.next_batch_token_file, "r") as f:
+                data = json.load(f)
+                self.client.next_batch = data["next_batch_token"]
+        except IOError:
+            print(f"Couldn't load next batch token from file.")
+        except json.JSONDecodeError:
+                print("Couldn't read JSON file; overwriting")
```

Comment:
I am not sure this is necessary. We can just make a second sync request or handle it without next batch?

Reply:
Alternatively cache this to redis?

Reply:
https://github.com/MatrixDoctors/matrix-cerberus/blob/804a5a7a08e061c93784962d48207ad52c715eb9/backend/app/core/sessions.py#L22-L46